pandas/tseries/index.py

```diff
@@ -77,16 +77,19 @@ def f(self):
 
             result = tslib.get_start_end_field(values, field, self.freqstr,
                                                month_kw)
+            result = self._maybe_mask_results(result, convert='float64')
+
```

Comment:
oh, these are a bug in ``_mask_missing_values`` then. It needs to ignore ``object`` and ``boolean`` dtypes.